Manage unofficial exporters

lagooj opened the pull request:
<!--
Please fill in this PR template to make it easier to review and merge.

It has been designed so that a lot of it can be completed *after* submitting it,
e.g. filling in the checklists.

Notes:
1. Please keep the PR as small as is practicable; the larger a PR gets, the harder it becomes to review and the more time it requires to get it merged.
2. Similarly, please avoid PRs that cover more than one type; it should be a _bug fix_ *OR* a _new feature_ *OR* a _refactor_, etc.
3. Please direct questions to the [`#formulas` channel on Slack](https://saltstackcommunity.slack.com/messages/C7LG8SV54/), which is bridged to `#saltstack-formulas` on Freenode.
4. Feel free to suggest improvements to this template by reporting an issue or submitting a PR.  The source of this template is:
  - https://github.com/saltstack-formulas/.github/blob/master/.github/pull_request_template.md
-->

### PR progress checklist (to be filled in by reviewers)
<!-- Please leave this checklist for reviewers to tick as they work through the PR. -->

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

### What type of PR is this?
Support "unofficial" exporter with archive as upstream

#### Primary type
<!-- There really should be only *one* of these types ticked for each PR. -->

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [X] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type
<!-- Most PRs should include all of the following types as well. -->

- [ ] `[docs]`     Documentation changes
- [X] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?
<!-- If so, change the following to a `Yes` and explain what the breaking changes are. -->
<!-- If there are multiple breaking changes, list them all. -->

No.

### Related issues and/or pull requests
<!-- Please link any related issues/PRs here, especially any issues that are closed by this PR. -->



### Describe the changes you're proposing
Taking care of "unofficial exporter" as postgresql, php-fpm ... 

### Pillar / config required to test the proposed changes
Under pkg:component:name:archive
adding 
official: true/false ( default => true)
tar: true/false ( default => true)


### Debug log showing how the proposed changes work


### Documentation checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [ ] Updated the `README` (e.g. `Available states`).
- [X] Updated `pillar.example`.

### Testing checklist
<!-- Please tick each box that is relevant (after creating the PR). -->

- [ ] Included in Kitchen (i.e. under `state_top`).
- [X] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [X] Updated the relevant test pillar.

### Additional context
<!-- Add any other context about the proposed changes here. -->

noelmcloughlin reviewed (changes requested):
LGTM mostly. I also need this feature.  
I am not sure `tar: boolean` is needed because the `archive` dict is always an archive.  So `file.managed` will work for both official and unofficial exporters.  There is probably no need to introduce `archive.extracted` as well since `file.managed` handles archives too? Just check the extra code is actually needed.

lagooj commented:
> LGTM mostly. I also need this feature.
> I am not sure `tar: boolean` is needed because the `archive` dict is always an archive. So `file.managed` will work for both official and unofficial exporters. There is probably no need to introduce `archive.extracted` as well since `file.managed` handles archives too? Just check the extra code is actually needed.
Hi,
I kept the original mechanism, to be sure to break nothing, as it's kind of trivial modification.
Not all opensource exporter deliver an archive, some as php-fpm provide only binary file, that's why I implemented tar: bool, in order to be able to switch to file managed if needed.
archive.extracted, extracts on fly, I didn't find the same behavior for file.managed in salt documentation, but I could be wrong :)

noelmcloughlin commented:
Get Travis CI passing and I'll merge. thanks @lagooj

saltstack-formulas-travis commented:
:tada: This PR is included in version 5.3.0 :tada:

The release is available on [GitHub release](https://github.com/saltstack-formulas/prometheus-formula/releases/tag/v5.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
